jxboot-resources/templates/dev-env.yaml

```diff
@@ -35,7 +35,7 @@ spec:
 {{- else }}
     helmTemplate: true
 {{- end }}
-    kubeProvider: gke
+    kubeProvider: "{{ .Values.cluster.provider }}"
```

Comment:
Should we give it a default value of `gke` for backwards compat?

Reply:
its more for diagnostics rather than a value we really need; so its probably fine if its empty. Though this PR should default it: https://github.com/jenkins-x/jenkins-x-boot-config/pull/75